### ISSUE TITLE
Add calculator node

### ISSUE THREAD

**Issue by ealmloff** [floneum/floneum#33]
This is one part of #27 

Floneum should have a calculator node that resolves an equation to a number. It should support:

- operators: `+`, `-`, `*`, `/`, `^`

This node should take in a string equation and output a number result:

```rust
use floneum_rust::*;

#[export_plugin]
/// calculates an equation
fn calculate(equation: String) -> f64 {
    todo!()
}
```

If possible, we should try to avoid using the python plugin because the bundle size of that plugin is quite large and we don't really need all of python's power here. We could use a parser like [pest](https://pest.rs/book/examples/calculator.html) or a library in rust that solves equations

Optional enhancements:
- [Structured generation](https://floneum.com/docs/user/concepts/structured) support/example for equations
- Algebraic equation simplification

**Comment by LafCorentin** [comment]
Do you mean i64 as output type or do you intend to add f64 to the availables outputs ? @ealmloff

**Comment by LafCorentin** [comment]
Other possibilites
[meval-rs](https://github.com/rekka/meval-rs)
[Mexprp](https://docs.rs/mexprp/latest/mexprp/)
[arithmetic_parser](https://docs.rs/arithmetic-parser/latest/arithmetic_parser/)
[math-parse](https://crates.io/crates/math-parse)
[exmex](https://lib.rs/crates/exmex)

**Comment by ealmloff** [comment]
> Do you mean i64 as output type or do you intend to add f64 to the availables outputs ? @ealmloff

We should add f64 to the primitive-values enum ([here](https://github.com/floneum/floneum/blob/5089944c3f36ea4668b144d1415509dd7daed7ac/floneum/wit/plugin.wit#L67-L81)).

Kalosm now supports tools including a [calculator tool](https://github.com/floneum/floneum/blob/5089944c3f36ea4668b144d1415509dd7daed7ac/interfaces/kalosm-language/src/tool/calculator.rs#L208). The logic for that tool may be useful when implementing the floneum node

**Comment by ealmloff** [comment]
Closed by #166
